gst/nnstreamer/nnstreamer_plugin_api_filter.h

```diff
@@ -184,41 +252,81 @@ typedef struct _GstTensorFilterFramework
        *
        * @param[in] hw backend accelerator hardware
        * @return 0 if supported. -errno if not supported.
        */
+    };
+
+    /**
+     * @brief Tensor_Filter Subplugin definition Version 2
+     */
+    struct /** _GstTensorFilterFramework_v2 */
+    {
+      int (*getFrameworkInfo) (void **private_data, GstTensorFilterFrameworkInfo *fw_info);
+      /**< Mandatory callback. Get the framworks statically determined info. Argument 'private_data' can be NULL. If provided 'private_data' is not NULL, then some info, such as 'allocate_in_invoke', can be updated based on the model being used (inferred from the 'private_data' provided). This updated info is useful for custom filter, as some custom filter's ability to support 'allocate_in_invoke' depends on the opened model.
+       *
+       * @param[in] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer. This parameter can be NULL.
+       * @param[out] fw_info struct to hold frameworks info. Must be allocated by the caller (return value).
+       * @return 0 if OK. non-zero if error.
+       */
+
+      int (*getModelInfo) (const GstTensorFilterProperties * prop,
+          void **private_data, model_info_ops ops, GstTensorsInfo *in_info,
+          GstTensorsInfo *out_info);
+      /**< Mandatory callback. Gets the model related tensor info.
+       * If ops == GET_INPUT_INFO, in_info would contain the input tensor info.
+       * If ops == GET_OUTPUT_INFO, out_info would contain the output tensor info.
+       * If ops == SET_INPUT_INFO, in_info would contain the provided input tensor info, and out_info would contain the updated output tensor info.
+       * If SET_INPUT_INFO operation is not supported, then both GET_INPUT_INFO and GET_OUTPUT_INFO must be supported.
+       *
+       * Note: Tensor_Filter::main will configure input dimension from pad-cap in
+       * run-time for the sub-plugin. Then, the sub-plugin is required to return
+       * corresponding output dimension, and will call SET_INPUT_INFO operation.
+       *
+       * Note: With SET_INPUT_INFO operation, the caller must NOT allocate or fix
+       * internal data structure based on the return value until invoke is called.
+       * Gstreamer may try different dimensions before settling down.
+       *
+       * @param[in] prop read-only property values
+       * @param[in/out] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer.
+       * @param[in] ops operation to be performed
+       * @param[in/out] in_info structure of input tensor info
+       * @param[out] out_info structure of output tensor info (return value)
+       * @return 0 if OK. non-zero if error. -ENOENT is operation is not supported.
+       */
+
+      int (*sendEvent) (const GstTensorFilterProperties * prop, void **private_data,
```

Comment:
Wouldn't ```eventHandler``` be more intuitive for developers?

Reply:
Updated.